### ISSUE TITLE
NullPointerException in SQLite Meta.migrateTo() when using DatabaseMetaData backed Meta containing FK referencing wrong identifier case

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#10741]
SQLite is a case insensitive database, and it does not normalise any cases of any identifiers.

Given a table like this:

```sql
CREATE TABLE t (
  ID INT NOT NULL,
  CONSTRAINT pk_t PRIMARY KEY (ID),
  CONSTRAINT fk_T_UPPER FOREIGN KEY (ID) REFERENCES T (ID),
  CONSTRAINT fk_t_lower FOREIGN KEY (ID) REFERENCES t (ID)
);
```

Notice the declaration of the table uses lower case `t` but it is referenced by `fk_T_UPPER` using an upper case `T`. The case is irrelevant, but preserved, meaning our lookups in `MetaImpl.MetaTable::getReferences` produce a NPE:

```
java.lang.NullPointerException
	at org.jooq.impl.MetaImpl$MetaTable.getReferences(MetaImpl.java:698)
	at org.jooq.impl.FilteredMeta$FilteredTable.getReferences(FilteredMeta.java:388)
	at org.jooq.impl.Diff$12.merge(Diff.java:330)
	at org.jooq.impl.Diff$12.merge(Diff.java:1)
	at org.jooq.impl.Diff.append(Diff.java:707)
	at org.jooq.impl.Diff.append(Diff.java:651)
	at org.jooq.impl.Diff.appendTables(Diff.java:354)
	at org.jooq.impl.Diff$3.merge(Diff.java:169)
	at org.jooq.impl.Diff$3.merge(Diff.java:1)
	at org.jooq.impl.Diff.append(Diff.java:707)
	at org.jooq.impl.Diff.append(Diff.java:651)
	at org.jooq.impl.Diff.appendSchemas(Diff.java:175)
	at org.jooq.impl.Diff$13.merge(Diff.java:124)
	at org.jooq.impl.Diff$13.merge(Diff.java:1)
	at org.jooq.impl.Diff.append(Diff.java:707)
	at org.jooq.impl.Diff.append(Diff.java:651)
	at org.jooq.impl.Diff.appendCatalogs(Diff.java:113)
	at org.jooq.impl.Diff.queries(Diff.java:109)
	at org.jooq.impl.AbstractMeta.migrateTo(AbstractMeta.java:509)
	at org.jooq.impl.AbstractMeta.migrateTo(AbstractMeta.java:504)
	at org.jooq.test.all.testcases.MetaDataTests.testMetaMigrateTo(MetaDataTests.java:162)
	at org.jooq.test.jOOQAbstractTest.testMetaMigrateTo(jOOQAbstractTest.java:2525)
```

Here's what the JDBC driver does:

```sql
SELECT *
FROM pragma_foreign_key_list('t') 
```

Resulting in:

```
id	seq	"table"	"from"	"to"	on_update	on_delete	"match"
0	0	t	ID	ID	NO ACTION	NO ACTION	NONE
1	0	T	ID	ID	NO ACTION	NO ACTION	NONE
```

----
See also:
- https://github.com/jOOQ/jOOQ/issues/10738 NullPointerException in Meta.migrateTo() when using DatabaseMetaData backed Meta containing views
- https://github.com/jOOQ/jOOQ/issues/2656 Add Settings to govern case insensitive identifier lookups

**Comment by lukaseder** [comment]
We can work around this particular problem for SQLite without having to go all in on case insensitive lookups (#2656).

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.14.0. Backport scheduled for 3.13.6 (#10742)

**Comment by lukaseder** [comment]
The current solution won't work if a locale specific ignore case comparison is needed.
